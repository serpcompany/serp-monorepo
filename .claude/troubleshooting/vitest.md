https://vitest.dev/guide/common-errors.html