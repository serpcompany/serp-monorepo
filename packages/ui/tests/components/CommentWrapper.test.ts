import { describe, expect, it } from 'vitest';
import CommentWrapper from '../../components/CommentWrapper.vue';
import ComponentRender from '../componentRender';
import '../mockUseUserSession';

describe('CommentWrapper Snapshot', () => {
  const baseProps = {
    module: 'test-module',
    id: 1,
    currentIndex: 0,
    initialMessageLimit: '10',
    maxLineLimit: '40',
    maxShowingDepth: '5',
    maxCommentLength: '1000',
    depthLength: 0,
    commentBackgroundColor: 'white',
    commentTextColor: '#1d2129',
    userNameColor: 'rgb(6, 177, 183)',
    wrapperSize: '1024',
    parentIds: [],
    parentIndices: [],
    comment: {
      id: 101,
      image: 'https://example.com/avatar.jpg',
      name: 'Test User',
      content: 'This is a test comment.\nIt has two lines.',
      timestamp: '1973-04-29T21:33:09Z',
      createdAt: '1973-04-29T21:33:09+00:00',
      updatedAt: '1973-04-29T21:33:09+00:00',
      replies: [],
      replyCount: 0,
      email: 'test@example.com',
      lineCount: 2
    }
  };
  it.each([
    ['default comment', { props: baseProps }],
    [
      'updated comment (with Updated badge)',
      {
        props: {
          ...baseProps,
          comment: {
            ...baseProps.comment,
            updatedAt: '1974-04-29T21:33:09+00:00'
          }
        }
      }
    ],
    [
      'comment with replies',
      {
        props: {
          ...baseProps,
          comment: {
            ...baseProps.comment,
            replies: [
              {
                id: 201,
                image: 'https://example.com/reply-avatar.jpg',
                name: 'Reply User',
                content: 'This is a reply.',
                timestamp: '1973-04-29T21:33:09Z',
                createdAt: '1973-04-29T21:33:09+00:00',
                updatedAt: '1973-04-29T21:33:09+00:00',
                replies: [],
                replyCount: 0,
                email: 'reply@example.com',
                lineCount: 1
              }
            ],
            replyCount: 1
          }
        }
      }
    ],
    [
      'comment with nested replies',
      {
        props: {
          ...baseProps,
          comment: {
            ...baseProps.comment,
            replies: [
              {
                id: 202,
                image: 'https://example.com/reply-avatar.jpg',
                name: 'Reply User',
                content: 'This is a reply.',
                timestamp: '1973-04-29T21:33:09Z',
                createdAt: '1973-04-29T21:33:09+00:00',
                updatedAt: '1973-04-29T21:33:09+00:00',
                replies: [
                  {
                    id: 203,
                    image: 'https://example.com/reply-avatar.jpg',
                    name: 'Reply User',
                    content: 'This is a reply.',
                    timestamp: '1973-04-29T21:33:09Z',
                    createdAt: '1973-04-29T21:33:09+00:00',
                    updatedAt: '1973-04-29T21:33:09+00:00',
                    replies: [],
                    replyCount: 0,
                    email: 'reply@example.com',
                    lineCount: 1
                  },
                  {
                    id: 204,
                    image: 'https://example.com/reply-avatar.jpg',
                    name: 'Reply User',
                    content: 'This is a reply.',
                    timestamp: '1973-04-29T21:33:09Z',
                    createdAt: '1973-04-29T21:33:09+00:00',
                    updatedAt: '1973-04-29T21:33:09+00:00',
                    replies: [],
                    replyCount: 0,
                    email: 'reply@example.com',
                    lineCount: 1
                  }
                ],
                replyCount: 2,
                email: 'reply@example.com',
                lineCount: 1
              }
            ],
            replyCount: 1
          }
        }
      }
    ]
  ])(
    'renders %s correctly',
    async (desc: string, options: { props: unknown }) => {
      const html = await ComponentRender(
        `CommentWrapper ${desc}`,
        options,
        CommentWrapper
      );
      expect(html).toMatchSnapshot();
    }
  );
});
