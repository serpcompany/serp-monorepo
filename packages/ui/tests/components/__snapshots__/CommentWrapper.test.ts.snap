// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CommentWrapper Snapshot > renders comment with nested replies correctly 1`] = `
<div class="comment-wrapper">
  <div class="wrapper">
    <div class="
      rounded-t-full
      w-12
    ">
      <img
        alt="Avatar"
        class="
          h-12
          p-1
          rounded-full
          w-12
        "
        data-nuxt-img=""
        src="https://example.com/avatar.jpg"
        srcset="https://example.com/avatar.jpg 1x, https://example.com/avatar.jpg 2x"
      />
    </div>
    <div class="comment-wrapper">
      <div class="name-wrapper">
        <div class="
          name
          text-(--ui-primary)
        ">
          Test User
        </div>
        <span class="dot">
          •
        </span>
        <span
          class="time"
          title="Thursday, January 1, 1970 at 9:00:01 AM"
        >
          just now
        </span>
        <!-- v-if -->
        <div
          class="
            expand
            hover:text-(--ui-primary)
          "
          style="display: none;"
        >
          <span title="Hide Comment">
            −
          </span>
        </div>
      </div>
      <div class="
        bg-white
        comment
        dark:bg-neutral-800
        dark:text-white
        text-black
      ">
        This is a test comment.
        It has two lines.
      </div>
      <div
        class="
          comment-box
          dark:text-white
          text-black
        "
        style="display: none;"
      >
        <div class="
          text-(--ui-primary)
          user-name
        ">
          Test User
        </div>
        <textarea
          aria-label="Add Comment"
          class="
            add-comment
            bg-white
            dark:bg-neutral-800
          "
          name="addUpdate"
          placeholder="Update message"
          spellcheck="false"
        ></textarea>
        <button
          aria-label="Update"
          class="
            bg-primary
            hover:bg-primary-800
          "
        >
          <span>
            Update
          </span>
        </button>
        <div class="
          bg-primary
          remaining-letter
        ">
          <span>
            1000
          </span>
        </div>
      </div>
      <div
        class="
          hover:text-(--ui-primary)
          reply
        "
        style="display: none;"
      >
        <div>
          Never mind
        </div>
      </div>
      <div class="reply">
        <div class="hover:text-(--ui-primary)">
          Reply
        </div>
        <span class="dot">
          •
        </span>
        <div class="hover:text-(--ui-primary)">
          Hide replies
        </div>
        <span class="dot">
          •
        </span>
        <div class="hover:text-(--ui-primary)">
          Update
        </div>
        <span class="dot">
          •
        </span>
        <div class="delete">
          <div class="delete-text">
            Delete
          </div>
          <!-- v-if -->
        </div>
      </div>
      <!-- v-if -->
      <transition-group-stub
        appear="true"
        css="true"
        name="fade"
        persisted="false"
        tag="div"
      >
        <div class="comment-wrapper">
          <div class="wrapper">
            <div class="
              rounded-t-full
              w-12
            ">
              <img
                alt="Avatar"
                class="
                  h-12
                  p-1
                  rounded-full
                  w-12
                "
                data-nuxt-img=""
                src="https://example.com/reply-avatar.jpg"
                srcset="https://example.com/reply-avatar.jpg 1x, https://example.com/reply-avatar.jpg 2x"
              />
            </div>
            <div class="comment-wrapper">
              <div class="name-wrapper">
                <div class="
                  name
                  text-(--ui-primary)
                ">
                  Reply User
                </div>
                <span class="dot">
                  •
                </span>
                <span
                  class="time"
                  title="Thursday, January 1, 1970 at 9:00:01 AM"
                >
                  just now
                </span>
                <!-- v-if -->
                <div
                  class="
                    expand
                    hover:text-(--ui-primary)
                  "
                  style="display: none;"
                >
                  <span title="Hide Comment">
                    −
                  </span>
                </div>
              </div>
              <div class="
                bg-white
                comment
                dark:bg-neutral-800
                dark:text-white
                text-black
              ">
                This is a reply.
              </div>
              <div
                class="
                  comment-box
                  dark:text-white
                  text-black
                "
                style="display: none;"
              >
                <div class="
                  text-(--ui-primary)
                  user-name
                ">
                  Reply User
                </div>
                <textarea
                  aria-label="Add Comment"
                  class="
                    add-comment
                    bg-white
                    dark:bg-neutral-800
                  "
                  name="addUpdate"
                  placeholder="Update message"
                  spellcheck="false"
                ></textarea>
                <button
                  aria-label="Update"
                  class="
                    bg-primary
                    hover:bg-primary-800
                  "
                >
                  <span>
                    Update
                  </span>
                </button>
                <div class="
                  bg-primary
                  remaining-letter
                ">
                  <span>
                    1000
                  </span>
                </div>
              </div>
              <div
                class="
                  hover:text-(--ui-primary)
                  reply
                "
                style="display: none;"
              >
                <div>
                  Never mind
                </div>
              </div>
              <div class="reply">
                <div class="hover:text-(--ui-primary)">
                  Reply
                </div>
                <span class="dot">
                  •
                </span>
                <div class="hover:text-(--ui-primary)">
                  Hide replies
                </div>
                <span class="dot">
                  •
                </span>
                <div class="hover:text-(--ui-primary)">
                  Update
                </div>
                <span class="dot">
                  •
                </span>
                <div class="delete">
                  <div class="delete-text">
                    Delete
                  </div>
                  <!-- v-if -->
                </div>
              </div>
              <!-- v-if -->
              <transition-group-stub
                appear="true"
                css="true"
                name="fade"
                persisted="false"
                tag="div"
              >
                <div class="comment-wrapper">
                  <div class="wrapper">
                    <div class="
                      rounded-t-full
                      w-12
                    ">
                      <img
                        alt="Avatar"
                        class="
                          h-12
                          p-1
                          rounded-full
                          w-12
                        "
                        data-nuxt-img=""
                        src="https://example.com/reply-avatar.jpg"
                        srcset="https://example.com/reply-avatar.jpg 1x, https://example.com/reply-avatar.jpg 2x"
                      />
                    </div>
                    <div class="comment-wrapper">
                      <div class="name-wrapper">
                        <div class="
                          name
                          text-(--ui-primary)
                        ">
                          Reply User
                        </div>
                        <span class="dot">
                          •
                        </span>
                        <span
                          class="time"
                          title="Thursday, January 1, 1970 at 9:00:01 AM"
                        >
                          just now
                        </span>
                        <!-- v-if -->
                        <div
                          class="
                            expand
                            hover:text-(--ui-primary)
                          "
                          style="display: none;"
                        >
                          <span title="Hide Comment">
                            −
                          </span>
                        </div>
                      </div>
                      <div class="
                        bg-white
                        comment
                        dark:bg-neutral-800
                        dark:text-white
                        text-black
                      ">
                        This is a reply.
                      </div>
                      <div
                        class="
                          comment-box
                          dark:text-white
                          text-black
                        "
                        style="display: none;"
                      >
                        <div class="
                          text-(--ui-primary)
                          user-name
                        ">
                          Reply User
                        </div>
                        <textarea
                          aria-label="Add Comment"
                          class="
                            add-comment
                            bg-white
                            dark:bg-neutral-800
                          "
                          name="addUpdate"
                          placeholder="Update message"
                          spellcheck="false"
                        ></textarea>
                        <button
                          aria-label="Update"
                          class="
                            bg-primary
                            hover:bg-primary-800
                          "
                        >
                          <span>
                            Update
                          </span>
                        </button>
                        <div class="
                          bg-primary
                          remaining-letter
                        ">
                          <span>
                            1000
                          </span>
                        </div>
                      </div>
                      <div
                        class="
                          hover:text-(--ui-primary)
                          reply
                        "
                        style="display: none;"
                      >
                        <div>
                          Never mind
                        </div>
                      </div>
                      <div class="reply">
                        <div class="hover:text-(--ui-primary)">
                          Reply
                        </div>
                        <!-- v-if -->
                        <span class="dot">
                          •
                        </span>
                        <div class="hover:text-(--ui-primary)">
                          Update
                        </div>
                        <span class="dot">
                          •
                        </span>
                        <div class="delete">
                          <div class="delete-text">
                            Delete
                          </div>
                          <!-- v-if -->
                        </div>
                      </div>
                      <!-- v-if -->
                      <transition-group-stub
                        appear="true"
                        css="true"
                        name="fade"
                        persisted="false"
                        tag="div"
                      ></transition-group-stub>
                      <!-- v-if -->
                    </div>
                  </div>
                </div>
                <div class="comment-wrapper">
                  <div class="wrapper">
                    <div class="
                      rounded-t-full
                      w-12
                    ">
                      <img
                        alt="Avatar"
                        class="
                          h-12
                          p-1
                          rounded-full
                          w-12
                        "
                        data-nuxt-img=""
                        src="https://example.com/reply-avatar.jpg"
                        srcset="https://example.com/reply-avatar.jpg 1x, https://example.com/reply-avatar.jpg 2x"
                      />
                    </div>
                    <div class="comment-wrapper">
                      <div class="name-wrapper">
                        <div class="
                          name
                          text-(--ui-primary)
                        ">
                          Reply User
                        </div>
                        <span class="dot">
                          •
                        </span>
                        <span
                          class="time"
                          title="Thursday, January 1, 1970 at 9:00:01 AM"
                        >
                          just now
                        </span>
                        <!-- v-if -->
                        <div
                          class="
                            expand
                            hover:text-(--ui-primary)
                          "
                          style="display: none;"
                        >
                          <span title="Hide Comment">
                            −
                          </span>
                        </div>
                      </div>
                      <div class="
                        bg-white
                        comment
                        dark:bg-neutral-800
                        dark:text-white
                        text-black
                      ">
                        This is a reply.
                      </div>
                      <div
                        class="
                          comment-box
                          dark:text-white
                          text-black
                        "
                        style="display: none;"
                      >
                        <div class="
                          text-(--ui-primary)
                          user-name
                        ">
                          Reply User
                        </div>
                        <textarea
                          aria-label="Add Comment"
                          class="
                            add-comment
                            bg-white
                            dark:bg-neutral-800
                          "
                          name="addUpdate"
                          placeholder="Update message"
                          spellcheck="false"
                        ></textarea>
                        <button
                          aria-label="Update"
                          class="
                            bg-primary
                            hover:bg-primary-800
                          "
                        >
                          <span>
                            Update
                          </span>
                        </button>
                        <div class="
                          bg-primary
                          remaining-letter
                        ">
                          <span>
                            1000
                          </span>
                        </div>
                      </div>
                      <div
                        class="
                          hover:text-(--ui-primary)
                          reply
                        "
                        style="display: none;"
                      >
                        <div>
                          Never mind
                        </div>
                      </div>
                      <div class="reply">
                        <div class="hover:text-(--ui-primary)">
                          Reply
                        </div>
                        <!-- v-if -->
                        <span class="dot">
                          •
                        </span>
                        <div class="hover:text-(--ui-primary)">
                          Update
                        </div>
                        <span class="dot">
                          •
                        </span>
                        <div class="delete">
                          <div class="delete-text">
                            Delete
                          </div>
                          <!-- v-if -->
                        </div>
                      </div>
                      <!-- v-if -->
                      <transition-group-stub
                        appear="true"
                        css="true"
                        name="fade"
                        persisted="false"
                        tag="div"
                      ></transition-group-stub>
                      <!-- v-if -->
                    </div>
                  </div>
                </div>
              </transition-group-stub>
              <!-- v-if -->
            </div>
          </div>
        </div>
      </transition-group-stub>
      <!-- v-if -->
    </div>
  </div>
</div>
`;

exports[`CommentWrapper Snapshot > renders comment with replies correctly 1`] = `
<div class="comment-wrapper">
  <div class="wrapper">
    <div class="
      rounded-t-full
      w-12
    ">
      <img
        alt="Avatar"
        class="
          h-12
          p-1
          rounded-full
          w-12
        "
        data-nuxt-img=""
        src="https://example.com/avatar.jpg"
        srcset="https://example.com/avatar.jpg 1x, https://example.com/avatar.jpg 2x"
      />
    </div>
    <div class="comment-wrapper">
      <div class="name-wrapper">
        <div class="
          name
          text-(--ui-primary)
        ">
          Test User
        </div>
        <span class="dot">
          •
        </span>
        <span
          class="time"
          title="Thursday, January 1, 1970 at 9:00:01 AM"
        >
          just now
        </span>
        <!-- v-if -->
        <div
          class="
            expand
            hover:text-(--ui-primary)
          "
          style="display: none;"
        >
          <span title="Hide Comment">
            −
          </span>
        </div>
      </div>
      <div class="
        bg-white
        comment
        dark:bg-neutral-800
        dark:text-white
        text-black
      ">
        This is a test comment.
        It has two lines.
      </div>
      <div
        class="
          comment-box
          dark:text-white
          text-black
        "
        style="display: none;"
      >
        <div class="
          text-(--ui-primary)
          user-name
        ">
          Test User
        </div>
        <textarea
          aria-label="Add Comment"
          class="
            add-comment
            bg-white
            dark:bg-neutral-800
          "
          name="addUpdate"
          placeholder="Update message"
          spellcheck="false"
        ></textarea>
        <button
          aria-label="Update"
          class="
            bg-primary
            hover:bg-primary-800
          "
        >
          <span>
            Update
          </span>
        </button>
        <div class="
          bg-primary
          remaining-letter
        ">
          <span>
            1000
          </span>
        </div>
      </div>
      <div
        class="
          hover:text-(--ui-primary)
          reply
        "
        style="display: none;"
      >
        <div>
          Never mind
        </div>
      </div>
      <div class="reply">
        <div class="hover:text-(--ui-primary)">
          Reply
        </div>
        <span class="dot">
          •
        </span>
        <div class="hover:text-(--ui-primary)">
          Hide replies
        </div>
        <span class="dot">
          •
        </span>
        <div class="hover:text-(--ui-primary)">
          Update
        </div>
        <span class="dot">
          •
        </span>
        <div class="delete">
          <div class="delete-text">
            Delete
          </div>
          <!-- v-if -->
        </div>
      </div>
      <!-- v-if -->
      <transition-group-stub
        appear="true"
        css="true"
        name="fade"
        persisted="false"
        tag="div"
      >
        <div class="comment-wrapper">
          <div class="wrapper">
            <div class="
              rounded-t-full
              w-12
            ">
              <img
                alt="Avatar"
                class="
                  h-12
                  p-1
                  rounded-full
                  w-12
                "
                data-nuxt-img=""
                src="https://example.com/reply-avatar.jpg"
                srcset="https://example.com/reply-avatar.jpg 1x, https://example.com/reply-avatar.jpg 2x"
              />
            </div>
            <div class="comment-wrapper">
              <div class="name-wrapper">
                <div class="
                  name
                  text-(--ui-primary)
                ">
                  Reply User
                </div>
                <span class="dot">
                  •
                </span>
                <span
                  class="time"
                  title="Thursday, January 1, 1970 at 9:00:01 AM"
                >
                  just now
                </span>
                <!-- v-if -->
                <div
                  class="
                    expand
                    hover:text-(--ui-primary)
                  "
                  style="display: none;"
                >
                  <span title="Hide Comment">
                    −
                  </span>
                </div>
              </div>
              <div class="
                bg-white
                comment
                dark:bg-neutral-800
                dark:text-white
                text-black
              ">
                This is a reply.
              </div>
              <div
                class="
                  comment-box
                  dark:text-white
                  text-black
                "
                style="display: none;"
              >
                <div class="
                  text-(--ui-primary)
                  user-name
                ">
                  Reply User
                </div>
                <textarea
                  aria-label="Add Comment"
                  class="
                    add-comment
                    bg-white
                    dark:bg-neutral-800
                  "
                  name="addUpdate"
                  placeholder="Update message"
                  spellcheck="false"
                ></textarea>
                <button
                  aria-label="Update"
                  class="
                    bg-primary
                    hover:bg-primary-800
                  "
                >
                  <span>
                    Update
                  </span>
                </button>
                <div class="
                  bg-primary
                  remaining-letter
                ">
                  <span>
                    1000
                  </span>
                </div>
              </div>
              <div
                class="
                  hover:text-(--ui-primary)
                  reply
                "
                style="display: none;"
              >
                <div>
                  Never mind
                </div>
              </div>
              <div class="reply">
                <div class="hover:text-(--ui-primary)">
                  Reply
                </div>
                <!-- v-if -->
                <span class="dot">
                  •
                </span>
                <div class="hover:text-(--ui-primary)">
                  Update
                </div>
                <span class="dot">
                  •
                </span>
                <div class="delete">
                  <div class="delete-text">
                    Delete
                  </div>
                  <!-- v-if -->
                </div>
              </div>
              <!-- v-if -->
              <transition-group-stub
                appear="true"
                css="true"
                name="fade"
                persisted="false"
                tag="div"
              ></transition-group-stub>
              <!-- v-if -->
            </div>
          </div>
        </div>
      </transition-group-stub>
      <!-- v-if -->
    </div>
  </div>
</div>
`;

exports[`CommentWrapper Snapshot > renders default comment correctly 1`] = `
<div class="comment-wrapper">
  <div class="wrapper">
    <div class="
      rounded-t-full
      w-12
    ">
      <img
        alt="Avatar"
        class="
          h-12
          p-1
          rounded-full
          w-12
        "
        data-nuxt-img=""
        src="https://example.com/avatar.jpg"
        srcset="https://example.com/avatar.jpg 1x, https://example.com/avatar.jpg 2x"
      />
    </div>
    <div class="comment-wrapper">
      <div class="name-wrapper">
        <div class="
          name
          text-(--ui-primary)
        ">
          Test User
        </div>
        <span class="dot">
          •
        </span>
        <span
          class="time"
          title="Thursday, January 1, 1970 at 9:00:01 AM"
        >
          just now
        </span>
        <!-- v-if -->
        <div
          class="
            expand
            hover:text-(--ui-primary)
          "
          style="display: none;"
        >
          <span title="Hide Comment">
            −
          </span>
        </div>
      </div>
      <div class="
        bg-white
        comment
        dark:bg-neutral-800
        dark:text-white
        text-black
      ">
        This is a test comment.
        It has two lines.
      </div>
      <div
        class="
          comment-box
          dark:text-white
          text-black
        "
        style="display: none;"
      >
        <div class="
          text-(--ui-primary)
          user-name
        ">
          Test User
        </div>
        <textarea
          aria-label="Add Comment"
          class="
            add-comment
            bg-white
            dark:bg-neutral-800
          "
          name="addUpdate"
          placeholder="Update message"
          spellcheck="false"
        ></textarea>
        <button
          aria-label="Update"
          class="
            bg-primary
            hover:bg-primary-800
          "
        >
          <span>
            Update
          </span>
        </button>
        <div class="
          bg-primary
          remaining-letter
        ">
          <span>
            1000
          </span>
        </div>
      </div>
      <div
        class="
          hover:text-(--ui-primary)
          reply
        "
        style="display: none;"
      >
        <div>
          Never mind
        </div>
      </div>
      <div class="reply">
        <div class="hover:text-(--ui-primary)">
          Reply
        </div>
        <!-- v-if -->
        <span class="dot">
          •
        </span>
        <div class="hover:text-(--ui-primary)">
          Update
        </div>
        <span class="dot">
          •
        </span>
        <div class="delete">
          <div class="delete-text">
            Delete
          </div>
          <!-- v-if -->
        </div>
      </div>
      <!-- v-if -->
      <transition-group-stub
        appear="true"
        css="true"
        name="fade"
        persisted="false"
        tag="div"
      ></transition-group-stub>
      <!-- v-if -->
    </div>
  </div>
</div>
`;

exports[`CommentWrapper Snapshot > renders updated comment (with Updated badge) correctly 1`] = `
<div class="comment-wrapper">
  <div class="wrapper">
    <div class="
      rounded-t-full
      w-12
    ">
      <img
        alt="Avatar"
        class="
          h-12
          p-1
          rounded-full
          w-12
        "
        data-nuxt-img=""
        src="https://example.com/avatar.jpg"
        srcset="https://example.com/avatar.jpg 1x, https://example.com/avatar.jpg 2x"
      />
    </div>
    <div class="comment-wrapper">
      <div class="name-wrapper">
        <div class="
          name
          text-(--ui-primary)
        ">
          Test User
        </div>
        <span class="dot">
          •
        </span>
        <span
          class="time"
          title="Thursday, January 1, 1970 at 9:00:01 AM"
        >
          just now
        </span>
        <!-- v-if -->
        <div
          class="
            expand
            hover:text-(--ui-primary)
          "
          style="display: none;"
        >
          <span title="Hide Comment">
            −
          </span>
        </div>
      </div>
      <div class="
        bg-white
        comment
        dark:bg-neutral-800
        dark:text-white
        text-black
      ">
        This is a test comment.
        It has two lines.
      </div>
      <div
        class="
          comment-box
          dark:text-white
          text-black
        "
        style="display: none;"
      >
        <div class="
          text-(--ui-primary)
          user-name
        ">
          Test User
        </div>
        <textarea
          aria-label="Add Comment"
          class="
            add-comment
            bg-white
            dark:bg-neutral-800
          "
          name="addUpdate"
          placeholder="Update message"
          spellcheck="false"
        ></textarea>
        <button
          aria-label="Update"
          class="
            bg-primary
            hover:bg-primary-800
          "
        >
          <span>
            Update
          </span>
        </button>
        <div class="
          bg-primary
          remaining-letter
        ">
          <span>
            1000
          </span>
        </div>
      </div>
      <div
        class="
          hover:text-(--ui-primary)
          reply
        "
        style="display: none;"
      >
        <div>
          Never mind
        </div>
      </div>
      <div class="reply">
        <div class="hover:text-(--ui-primary)">
          Reply
        </div>
        <!-- v-if -->
        <span class="dot">
          •
        </span>
        <div class="hover:text-(--ui-primary)">
          Update
        </div>
        <span class="dot">
          •
        </span>
        <div class="delete">
          <div class="delete-text">
            Delete
          </div>
          <!-- v-if -->
        </div>
      </div>
      <!-- v-if -->
      <transition-group-stub
        appear="true"
        css="true"
        name="fade"
        persisted="false"
        tag="div"
      ></transition-group-stub>
      <!-- v-if -->
    </div>
  </div>
</div>
`;
